packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -629,3 +772,194 @@ class PageTransitionsTheme with Diagnosticable {
+    ..color = Color.fromRGBO(0, 0, 0, opacity);
+  final double logicalWidth = image.width / pixelRatio;
+  final double logicalHeight = image.height / pixelRatio;
+  final double scaledLogicalWidth = logicalWidth * scale;
+  final double scaledLogicalHeight = logicalHeight * scale;
+  final double left = (logicalWidth - scaledLogicalWidth) / 2;
+  final double top = (logicalHeight - scaledLogicalHeight) / 2;
+  final Rect dst = Rect.fromLTWH(left, top, scaledLogicalWidth, scaledLogicalHeight);
+  context.canvas.drawImageRect(image, Rect.fromLTWH(0, 0, image.width.toDouble(), image.height.toDouble()), dst, paint);
+}
+
+void _updateScaledTransform(Matrix4 transform, double scale, Size size) {
+  transform.setIdentity();
+  if (scale == 1.0) {
+    return;
+  }
+  transform.setDiagonal(Vector4(scale, scale, 1, 1));
+  final double dx = (size.width - (size.width * scale)).abs() / 2;
+  final double dy = (size.height - (size.height * scale)).abs() / 2;
+  transform.translate(dx, dy);
+}
+
+mixin _ZoomTransitionBase {
+  // Don't rasterize if:
+  // 1. Rasterization is disabled by the platform.
+  // 2. The animation is paused/stopped.
+  // 3. The values of the scale/fade transition do not
+  //    benefit from rasterization.
+  final RasterWidgetController controller = RasterWidgetController();
+
+  late Animation<double> fadeTransition;
+  late Animation<double> scaleTransition;
+
+  bool get allowRasterization;
+
+  void onAnimationValueChange() {
+    if ((scaleTransition.value == 1.0) &&
+        (fadeTransition.value == 0.0 ||
+         fadeTransition.value == 1.0)) {
+        controller.rasterize = false;
+      } else {
+        controller.rasterize = allowRasterization;
+      }
+  }
+
+  void onAnimationStatusChange(AnimationStatus status) {
+    switch (status) {
+      case AnimationStatus.dismissed:
+      case AnimationStatus.completed:
+        controller.rasterize = false;
+        break;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        controller.rasterize = allowRasterization;
+        break;
+    }
+  }
+}
+
+
+class _ZoomEnterTransitionDelegate extends RasterWidgetDelegate implements RasterWidgetFallbackDelegate {
+  _ZoomEnterTransitionDelegate({
+    required this.reverse,
+    required this.scale,
+    required this.fade,
+    required this.animation,
+  }) {
+    animation.addListener(notifyListeners);
+    scale.addListener(notifyListeners);
+    fade.addListener(notifyListeners);
+  }
+
+  final bool reverse;
+  final Animation<double> animation;
+  final Animation<double> scale;
+  final Animation<double> fade;
+
+  final Matrix4 _transform = Matrix4.zero();
+  OpacityLayer? _oldOpacity;
+  TransformLayer? _oldTransform;
+
+  void _drawScrim(PaintingContext context, Offset offset, Size size) {
+    double scrimOpacity = 0.0;
+    // The transition's scrim opacity only increases on the forward transition.
+    // In the reverse transition, the opacity should always be 0.0.
```

Comment:
can this be asserted in case it ever changes?

Reply:
I added an assert to singular place this is used